### ISSUE TITLE
"defer rc.Close() or similar" check doesn't detect bad calls of Close() on nested fields.

### ISSUE THREAD

**Issue by dmitshur** [dominikh/go-staticcheck#16]
One of the checks in README says:

> - Don't `defer rc.Close()` before having checked the error returned by `Open` or similar.

The "or similar" statement implies it should work for detecting a similar error for `http.Get` and `defer resp.Body.Close()`:

```Go
package main

import (
	"fmt"
	"io"
	"io/ioutil"
	"net/http"
)

func main() {
	resp, err := http.Get("https://www.example.com/")
	defer resp.Body.Close()
	if err != nil {
		fmt.Println(err)
		return
	}
	// read resp.Body or something
	io.Copy(ioutil.Discard, resp.Body)
}
```

Running `staticcheck` produces zero output:

```
$ staticcheck
```

I've already let @dominikh know about this, but reporting it here for posterity and tracking, and so that I can close an open window and delete some files.

Related to golang/go#17780.
